### PR TITLE
refactor(server): auth service

### PR DESCRIPTION
**Pull request by jrasm91**
The PR moves auth, oauth, and immich-jwt (=> AuthCore) to domain.

### CODE REVIEW

File: server/libs/domain/src/auth/auth.core.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    } else {
+      accessTokenCookie = `${IMMICH_ACCESS_COOKIE}=${loginResponse.accessToken}; HttpOnly; Path=/; Max-Age=${maxAge}; SameSite=Strict;`;
+      authTypeCookie = `${IMMICH_AUTH_TYPE_COOKIE}=${authType}; HttpOnly; Path=/; Max-Age=${maxAge}; SameSite=Strict;`;
+    }
+    return [accessTokenCookie, authTypeCookie];
+  }
+
+  public async createLoginResponse(user: UserEntity, authType: AuthType, isSecure: boolean) {
+    const payload: JwtPayloadDto = { userId: user.id, email: user.email };
+    const accessToken = await this.generateToken(payload);
```
Comment (alextran1502) [inline server/libs/domain/src/auth/auth.core.ts:47]: `generateToken` is not an async method, I believe we can remove await here.

File: server/libs/domain/src/system-config/system-config.initial.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const INITIAL_SYSTEM_CONFIG = 'INITIAL_SYSTEM_CONFIG';
```
Comment (alextran1502) [inline server/libs/domain/src/system-config/system-config.initial.ts:1]: What is your plan for this file, what are additional information we will be using in this file?

**Comment by alextran1502** [comment]
Looks good, just have two comments. Thanks @jrasm91 for another quality PR

**Review by alextran1502 — APPROVED** [review]

File: server/libs/domain/src/system-config/system-config.initial.ts (hunk 2 above)
Comment (jrasm91) [reply server/libs/domain/src/system-config/system-config.initial.ts]: Just needed to put it somewhere else to not have a circular dependency, I merged it into the datetime file and renamed to `*.constants.ts*`

**Pull request merged**